addons/product_expiry/models/stock_move_line.py

```diff
@@ -48,3 +48,13 @@ def _onchange_product_id(self):
     def _assign_production_lot(self, lot):
         super()._assign_production_lot(lot)
         self.lot_id._update_date_values(self[0].expiration_date)
+
+    def _get_value_production_lot(self):
+        res = super()._get_value_production_lot()
+        res.update({
+            'expiration_date': self.expiration_date,
+            'use_date': False if(self.product_id.use_time == 0) else self.expiration_date - datetime.timedelta(days=(self.product_id.expiration_time - self.product_id.use_time)),
```

Comment:
```suggestion
            'use_date': self.product_id.use_time and self.expiration_date - datetime.timedelta(days=(self.product_id.expiration_time - self.product_id.use_time)),
```
(same for next fields)
Also, could you add (or edit an existing) test to check theses values ?